app/database/models/ms_schema/tasks_list.py

```diff
@@ -0,0 +1,210 @@
+
+    id = db.Column(db.Integer, primary_key=True)
+    tasks = db.Column(JsonCustomType)
+    next_task_id = db.Column(db.Integer)
+
+    def __init__(self, tasks: "TasksListModel" = None):
+        """Initializes tasks.
+
+        Args:
+            tasks: A list of tasks.
+        
+        Raises:
+            A Value Error if the task is not initialized.
+        """
+
+        if tasks is None:
+            self.tasks = []
+            self.next_task_id = 1
+        else:
+            if isinstance(tasks, list):
+                self.tasks = []
+                self.next_task_id = len(tasks) + 1
+            else:
+                raise ValueError(TypeError)
+
+    def add_task(
+        self, description: str, created_at: date, is_done=False, completed_at=None
+    ) -> None:
+        """Adds a task to the list of tasks.
+        
+        Args:
+            description: A description of the task added.
+            created_at: Date on which the task is created.
+            is_done: Boolean specifying completion of the task.
+            completed_at: Date on which task is completed.
+        """
+
+        task = {
+            TasksFields.ID.value: self.next_task_id,
+            TasksFields.DESCRIPTION.value: description,
+            TasksFields.IS_DONE.value: is_done,
+            TasksFields.CREATED_AT.value: created_at,
+            TasksFields.COMPLETED_AT.value: completed_at,
+        }
+        self.next_task_id += 1
+        self.tasks = self.tasks + [task]
+
+    def delete_task(self, task_id: int) -> None:
+        """Deletes a task from the list of tasks.
+
+        Args:
+            task_id: Id of the task to be deleted.
+        """
+
+        new_list = list(
+            filter(lambda task: task[TasksFields.ID.value] != task_id, self.tasks)
+        )
+
+        self.tasks = new_list
+        self.save_to_db()
+
+    def update_task(
+        self,
+        task_id: int,
+        description: str = None,
+        is_done: bool = None,
+        completed_at: date = None,
+    ) -> None:
+        """Updates a task.
+        
+        Args:
+            task_id: Id of the task to be updated.
+            description: A description of the task.
+            created_at: Date on which the task is created.
+            is_done: Boolean specifying completion of the task.
+            completed_at: Date on which task is completed.
+        """
+
+        new_list = []
+        for task in self.tasks:
+            if task[TasksFields.ID.value] == task_id:
+                new_task = task.copy()
+                if description is not None:
+                    new_task[TasksFields.DESCRIPTION.value] = description
+
+                if is_done is not None:
+                    new_task[TasksFields.IS_DONE.value] = is_done
+
+                if completed_at is not None:
+                    new_task[TasksFields.COMPLETED_AT.value] = completed_at
+
+                new_list = new_list + [new_task]
```

Comment:
+=?

Reply:
yeah,, not sure why this was approve before in mentorship too... 🤣